tests/test_tpp_backend.py

```diff
@@ -2983,3 +2985,51 @@ def test_dynamic_index_dates_with_invalid_expression():
                 on_or_before="last_day_of_month(earliest_bar) + 1 mnth",
             ),
         )
+
+
+def test_high_cost_drugs():
+    session = make_session()
+    session.add_all(
+        [
+            Patient(
+                HighCostDrugs=[
+                    HighCostDrugs(
+                        DrugName="foo", FinancialYear="201920", FinancialMonth="2"
+                    ),
+                    HighCostDrugs(
+                        DrugName="bar", FinancialYear="201920", FinancialMonth="6"
+                    ),
+                    HighCostDrugs(
+                        DrugName="foo", FinancialYear="201920", FinancialMonth="10"
+                    ),
+                ],
+            ),
+        ]
+    )
+    session.commit()
+    study = StudyDefinition(
+        population=patients.all(),
+        latest_drug_date=patients.with_high_cost_drugs(
+            find_last_match_in_period=True,
+            returning="date",
+            date_format="YYYY-MM",
+        ),
+        latest_bar_date=patients.with_high_cost_drugs(
+            drug_name_matches="bar",
+            find_last_match_in_period=True,
+            returning="date",
+            date_format="YYYY-MM",
+        ),
```

Comment:
This is fine but since there is only one `bar` event, the test will pass if you change `find_last_match_in_period` to `find_first_match_in_period`.

Reply:
Good point. I've made the tests slightly more extensive.